File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        ones = True
        for i in range(len(nums)):
            if(nums[i] == 0):
                ones = False
                break
        if(ones): return len(nums)-1
        prev = curr = max_len = 0;
        for i in range(len(nums)):
            if(nums[i] == 0):
                max_len = max(max_len, prev+curr)
                prev = curr
                curr = 0
            else:
                curr+=1
        return max(max_len, prev+curr)

File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int prev = 0, curr=0, longest=0;
        bool ones = true;
        for(int i=0; i<nums.size(); i++){
            if(nums[i] == 0){
                ones = false;
                break;
            } 
        }
        if(ones) return nums.size()-1;
        for(int i=0; i<nums.size(); i++){
            if(nums[i]==0){
                longest = max(longest, prev+curr);
                prev = curr;
                curr = 0;
            }else curr++;
        }
        return max(longest, prev+curr);
        
    }
};